app/controllers/api/functions.php

```diff
@@ -2045,6 +2046,73 @@
         $response->dynamic($execution, Response::MODEL_EXECUTION);
     });
 
+App::delete('/v1/functions/:functionId/executions/:executionId')
+    ->groups(['api', 'functions'])
+    ->desc('Delete execution')
+    ->label('scope', 'execution.write')
+    ->label('event', 'functions.[functionId].executions.[executionId].delete')
+    ->label('audits.event', 'executions.delete')
+    ->label('audits.resource', 'function/{request.functionId}')
+    ->label('sdk.auth', [APP_AUTH_TYPE_KEY])
+    ->label('sdk.namespace', 'functions')
+    ->label('sdk.method', 'deleteExecution')
+    ->label('sdk.description', '/docs/references/functions/delete-execution.md')
+    ->label('sdk.response.code', Response::STATUS_CODE_NOCONTENT)
+    ->label('sdk.response.model', Response::MODEL_NONE)
+    ->param('functionId', '', new UID(), 'Function ID.')
+    ->param('executionId', '', new UID(), 'Execution ID.')
+    ->inject('response')
+    ->inject('dbForProject')
+    ->inject('dbForConsole')
+    ->inject('queueForEvents')
+    ->action(function (string $functionId, string $executionId, Response $response, Database $dbForProject, Database $dbForConsole, Event $queueForEvents) {
+        $function = $dbForProject->getDocument('functions', $functionId);
+
+        if ($function->isEmpty()) {
+            throw new Exception(Exception::FUNCTION_NOT_FOUND);
+        }
+
+        $execution = $dbForProject->getDocument('executions', $executionId);
+        if ($execution->isEmpty()) {
+            throw new Exception(Exception::EXECUTION_NOT_FOUND);
+        }
+
+        if ($execution->getAttribute('functionId') !== $function->getId()) {
+            throw new Exception(Exception::EXECUTION_NOT_FOUND);
+        }
+        $status = $execution->getAttribute('status');
+
+        if (!in_array($status, ['completed', 'failed', 'scheduled'])) {
+            throw new Exception(Exception::EXECUTION_IN_PROGRESS);
+        }
+
+        if (!$dbForProject->deleteDocument('executions', $execution->getId())) {
+            throw new Exception(Exception::GENERAL_SERVER_ERROR, 'Failed to remove execution from DB');
+        }
+
+        if ($status === 'scheduled') {
+            $schedule = $dbForConsole->findOne('schedules', [
+                Query::equal('resourceId', [$execution->getId()]),
+                Query::equal('resourceType', [ScheduleExecutions::getSupportedResource()]),
+                Query::equal('active', [true]),
+            ]);
+
+            if ($schedule && !$schedule->isEmpty()) {
+                $schedule
+                    ->setAttribute('resourceUpdatedAt', DateTime::now())
+                    ->setAttribute('active', false);
+
+                Authorization::skip(fn () => $dbForConsole->updateDocument('schedules', $schedule->getId(), $schedule));
```

Comment:
Doesn't it get deleted during maintenance?

Reply:
Correct, we delete it in the scheduler. If we deleted  it here, and the scheduler already knew about the document, it would execute it.
So instead, we just update document to be in-active, and next time scheduler pulls updated, he will remove it from his list and delete document: https://github.com/appwrite/appwrite/blob/fef22825fb0d384490197aa9aaae3ec8da40dad0/src/Appwrite/Platform/Tasks/ScheduleExecutions.php#L31-L39